src/lib/mongoBackend/safeBsonGet.cpp

```diff
@@ -104,11 +104,18 @@ int getIntField(const BSONObj& b, const std::string& field)
 *
 * getLongField -
 */
-long getLongField(const BSONObj& b, const std::string& field)
+long long getLongField(const BSONObj& b, const std::string& field)
```

Comment:
This is the exact function you've been asking for.
I am not going to implement this part in the 10 times where it is called. That is a really bad idea.
Perhaps you just don't like the name of the function?